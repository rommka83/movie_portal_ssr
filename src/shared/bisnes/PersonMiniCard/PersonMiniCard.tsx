import React, { HTMLAttributes, PropsWithChildren } from 'react';
import styles from './personcard.module.css';
import { IPerson } from 'shared/types/IPerson';
import Image from 'next/image';
import classNames from 'classnames';
import Link from 'next/link';

interface IProps {
  person: IPerson;
  position?: boolean;
}

export function PersonMiniCard({
  person,
  position,
  className,
}: PropsWithChildren<HTMLAttributes<HTMLDivElement> & IProps>) {
  const personName = person.name ?? person.enName;
  if (!personName) return null;
  const [name] = personName.split(' ').slice(0, 1);
  const [surName] = personName.split(' ').slice(1);

  return (
    <Link href={`/ActorPage/${person.id}`} className={styles.link}>
      <div className={classNames(styles.root, className)}>
        {person.photo != null ? (
          <Image width={60} height={60} src={person.photo} alt={personName} className={styles.pic} />
        ) : (
          <div className={styles.noPhoto}>
            <span className={classNames(styles.noPhotoIcon, 'icon-person_56__0')} />
          </div>
        )}
        <p className={styles.name}>{name}</p>
        <p className={classNames(styles.name, styles.surName)}>{surName}</p>
        {position && <p className={styles.position}>{person.profession}</p>}
      </div>
    </Link>
  );
}
