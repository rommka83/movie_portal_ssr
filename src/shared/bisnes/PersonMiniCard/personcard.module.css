.root:hover .name {
  opacity: 1;
}
.root:hover .position {
  opacity: 1;
}
.root:hover .pic {
  transform: scale(1.05);
}
.link {
  width: 100%;
}
.pic {
  width: 100%;
  height: auto;
  margin-bottom: 10px;
  aspect-ratio: 1/1;
  object-fit: cover;
  border-radius: 50%;
  overflow: hidden;
  transition: transform 0.3s ease-in-out;
}
.noPhoto {
  display: flex;
  align-items: center;
  justify-content: center;

  width: 100%;
  aspect-ratio: 1/1;
  border-radius: 50%;

  background-color: var(--gray-60);
  font-size: 40px;
  color: var(--pink);
}

.name,
.surName {
  font-size: 13px;
  line-height: 16px;
  font-weight: 500;
  color: var(--gray-7e0);
  text-align: center;
  opacity: 0.5;
  transition: opacity 0.3s ease-in-out;
}
.surName {
  overflow: hidden;
  text-overflow: ellipsis;
}
.position {
  font-size: 13px;
  line-height: 16px;
  font-weight: 400;
  color: var(--gray-light);
  text-align: center;
  opacity: 0.5;
  transition: opacity 0.3s ease-in-out;
}
