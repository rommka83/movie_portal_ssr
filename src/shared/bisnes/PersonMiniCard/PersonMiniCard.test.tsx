import renderer from 'react-test-renderer';
import { PersonMiniCard } from './PersonMiniCard';

const person = {
  id: 1,
  photo: '',
  name: 'string',
  enName: 'string',
  description: 'string',
  profession: 'string',
  enProfession: 'string',
};

describe('проверка PersonMiniCard', () => {
  test('отрисовка PersonMiniCard ', () => {
    const component = renderer.create(<PersonMiniCard person={person} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
