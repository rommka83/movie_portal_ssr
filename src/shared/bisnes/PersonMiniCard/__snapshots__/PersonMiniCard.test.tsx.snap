// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка PersonMiniCard отрисовка PersonMiniCard  1`] = `
<a
  className="link"
  href="/ActorPage/1"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="root"
  >
    <img
      alt="string"
      className="pic"
      data-nimg="1"
      decoding="async"
      height={60}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src=""
      style={
        Object {
          "color": "transparent",
        }
      }
      width={60}
    />
    <p
      className="name"
    >
      string
    </p>
    <p
      className="name surName"
    />
  </div>
</a>
`;
